### PR TITLE
docker-machine: Deprecate both plugin and completion file

### LINKED ISSUE #1744: Deprecate Docker-Machine plugin and completion
Docker-Machine is deprecated, the recommendation is to move to native Docker apps on Windows/Mac. This might be a candidate for deprecation...

_Originally posted by @nwinkler in https://github.com/Bash-it/bash-it/pull/1741#discussion_r543158995_

### LINKED ISSUE #1786: Logging warning and not error when deprecating component
Errors logs are logs that should be reported to us, while warnings are warnings to the users. Deprecation logs are for the user only, so I think we should convert all deprecation error to deprecation warnings

### PR DESCRIPTION
**Pull request by NoahGorny**
Closes #1744
Closes #1786

## Description

Deprecate both plugin and completion file of docker-machine, as the command is deprecated as well.

## Motivation and Context

See #1744 for context

## How Has This Been Tested?

enabled locally, saw the error log

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] If my change requires a change to the documentation, I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] If I have added a new file, I also added it to ``clean_files.txt`` and formatted it using ``lint_clean_files.sh``.
- [ ] I have added tests to cover my changes, and all the new and existing tests pass.

### CODE REVIEW

File: completion/available/docker-machine.completion.bash [hunk 1]
```diff
@@ -1,431 +1,3 @@
...
-    if declare -F "${completion_func}" > /dev/null; then
-        ${completion_func}
-    fi
-
-    return 0
-}
-
-complete -F _docker_machine docker-machine docker-machine.exe
+# shellcheck shell=bash
+_log_error '"docker-machine" is now deprecated, and as such the bash completion for it is also deprecated.
```
Comment (nwinkler) [inline completion/available/docker-machine.completion.bash:2]: Don't know if we discussed this before, but do we want to raise the deprecation as an error? Wouldn't a warning be sufficient?

Comment (NoahGorny) [reply completion/available/docker-machine.completion.bash:2]: Hmm, I am unsure. I think errors are things that should be reported to us, while warning are warnings to the user, so I guess this really should just be log_warning. I say we merge this, and I will open an issue about it

Comment (nwinkler) [reply completion/available/docker-machine.completion.bash:2]: I think we both agree that it should be a warning then?

Comment (NoahGorny) [reply completion/available/docker-machine.completion.bash]: I updated the PR with a commit that changes all places like this

**Review by cornfeedhobo — APPROVED** [review]

**Pull request merged**